synapse/rest/client/v2_alpha/register.py

```diff
@@ -77,24 +77,33 @@ def on_POST(self, request):
         body = parse_json_object_from_request(request)
 
         assert_params_in_dict(
-            body, ["id_server", "client_secret", "email", "send_attempt"]
+            body, ["client_secret", "email", "send_attempt"]
         )
 
+        # Extract params from body
+        client_secret = body["client_secret"]
+        email = body["email"]
+        send_attempt = body["send_attempt"]
+        next_link = body.get("next_link")  # Optional param
+
         if not check_3pid_allowed(self.hs, "email", body["email"]):
             raise SynapseError(
                 403,
                 "Your email domain is not authorized to register on this server",
                 Codes.THREEPID_DENIED,
             )
 
-        existingUid = yield self.hs.get_datastore().get_user_id_by_threepid(
+        existing_user_id = yield self.hs.get_datastore().get_user_id_by_threepid(
             "email", body["email"]
         )
 
-        if existingUid is not None:
+        if existing_user_id is not None:
             raise SynapseError(400, "Email is already in use", Codes.THREEPID_IN_USE)
 
-        ret = yield self.identity_handler.requestEmailToken(**body)
+        ret = yield self.identity_handler.requestEmailToken(
+            None, email, client_secret, send_attempt, next_link
```

Comment:
`/register` isn't going to allow the client to specify the `id_server` any more, so don't pass it on.